로그인, 회원가입페이지 구현

seongjin0320 opened the pull request:
# 로그인, 회원가입페이지 구현

## 설명

- 로그인, 회원가입 기능 구현

- 로그인 UI 구성했습니다
- 로그인 UI, 비즈니스로직, 유효성검사스키마 각각 LoginPage.tsx, useLoginForm.ts, loginValidation.ts 파일로 분리했습니다.
- 회원가입 UI를 로그인 UI와 일관성있게 구성했습니다
- 회원가입 UI, 비즈니스로직, 유효성검사스키마 각각 RegisterPage.tsx, useRegisterForm.ts, registerValidation.ts 파일로 분리했습니다.
- 로그인, 회원가입 API(POST)는 아직 구현하지 않았습니다.

- **기타 정보**:

## 관련 이슈
로그인, 회원가입 페이지 API 구현 

<!--- 바로 아래에 이슈 링크를 추가해 주세요.-->
#9

seongjin0320 committed:
test: push test

seongjin0320 committed:
chore: react-hook-form 설치

seongjin0320 committed:
feat: 로그인페이지 UI 구성, 로그인 폼 구현

- 로그인 UI 구성
- 로그인 폼 react-hook-form 라이브러리 이용 구현
- LoginPage.tsx에서는 UI로직을 담당, useLoginForm.ts파일로 폼 로직 분리

seongjin0320 committed:
chore: zod 라이브러리 설치

- react-hook-form의 유효성 검증 로직을 위한 라이브러

seongjin0320 committed:
feat: 로그인 유효성 검사 로직 추가

- 유효성 검사 로직을 zod라이브러리 이용하여 구현했습니다
- 유효성 검사 로직은 백엔드와 상의 후에 변경해야할 수 있습니다
- 유효성이 검증되지 않으면 로그인버튼이 활성화되지 않는 기능 추가했습니다
- 로그인 이후에는 홈으로 navigate합니다

seongjin0320 committed:
refact: 인풋필드 공통컴포넌트 추가

- 로그인 인풋필드 앞으로 여러군데에서 사용할 것 같아서(회원가입 등) 공통컴포넌트로 정의했습니다

seongjin0320 committed:
feat: 회원가입 기능 추가

- 회원가입페이지 UI 로그인페이지와 일관성 있게 구현했습니다
- 회원가입페이지 UI는 RegisterPage.tsx에서 다루고, 폼 로직은 useResigterForm.ts에서 다루고, registerValidtaion.ts에서 유효성 로직을 다룹니다
- 회원가입 제출시에 API호출 로직은 구현되지 않았습니다
- 로그인페이지와 마찬가지로 react-hook-form, zod 라이브러리 사용했습니다

Moderator11 commented on src/pages/login/utils/loginValidation.ts:
```diff
@@ -0,0 +1,11 @@
+import { z } from 'zod';
+//유효성 검사 로직은 백엔드와 상의 후 변경해야할 수 있음.
```
유효성 룰은 인증/인가 부분 담당 백엔드분이 프론트쪽에서 정해도 된다고 하셨어요!

비밀번호 유효성 규칙에 대해서 보안적인 관점에서 이야기를 해보자면 다른 상용 서비스들과 비교해 비밀번호 로직이 조금 단순한 것 같습니다! 이러한 경우 Brute-Force 공격에 취약한데, 대부분 대소문자, 특수문자, 숫자를 혼용하여 비밀번호 조합을 만드는 것을 권장합니다.

하지만 이러한 규칙이 불필요하다고 보는 관점도 있습니다. 복잡한 규칙을 넣어봤자 사람들은 기억하기 쉽게 Mypassword1! 와 비슷한 형태로 만들기 때문입니다.

사용자 입장에서는 유효성 규칙 수가 적은 것이 좋긴 합니다! 그래서 유효성 규칙을 그대로 두는 것도 좋고, 보안성을 위해 유효성 규칙을 추가해도 좋을 것 같아요! 이 부분에 대해서는 다른 분들의 의견을 여쭤보고 싶습니다.
@seongjin0320 @whyyhyh

whyyhyh replied:
저는 저희 서비스가 개인만의 스페이스를 중요시 한다기 보단, 동시 수정에 중점을 둔 공유 플랫폼이기 때문에 비밀번호 보안은 널널하게 해도 괜찮다고 생각합니다!

seongjin0320 replied:
저는 저희 서비스가 금융이나, 예약 등의 서비스를 다루지 않기 때문에 여권번호 등의 민감한 정보를 다루지않아 보안상으로 크게 엄격하지는 않아도 된다고 생각합니다!  
하지만 필요할 때 바로 유효성 검사 강화할수있도록 주석으로 추가하겠습니다!

Moderator11 commented on src/pages/login/LoginPage.tsx:
```diff
@@ -1,5 +1,121 @@
+          error={errors.password}
+        />
+        <SignupGuide>
+          아직 계정이 없으신가요?{' '}
```
{' '} 이 부분은 prettier 규칙에 의해 자동 삽입된 코드인가요?

seongjin0320 replied:
아직 계정이 없으신가요? 와 회원가입 사이에 공백을 추가하기 위한 코드입니다 자동삽입된건 아닙니다

Moderator11 commented on src/pages/register/utils/registerValidation.ts:
```diff
@@ -0,0 +1,38 @@
+      .email({ message: '유효한 이메일 형식이 아닙니다' }),
+    name: z.string().min(1, { message: '이름을 입력해주세요' }),
+    password: z
+      .string()
+      .min(8, { message: '비밀번호는 8자 이상이어야 합니다' })
+      .regex(/^(?=.*[a-zA-Z])(?=.*[0-9])/, {
+        message: '비밀번호는 영문, 숫자를 모두 포함해야 합니다',
+      }),
```
로그인에서 비밀번호 유효성 규칙과 회원가입에서 비밀번호 유효성 규칙이 서로 다르네요. 회원가입에서 영문, 숫자를 모두 포함해야 한다는 규칙이 하나 더 있어요! 이 차이는 의도하신 부분인가요?

seongjin0320 replied:
의도한것은 아니긴했는데, 생각해보니 실제 다른 서비스들에서도 로그인시에 비밀번호의 자세한 유효성 피드백을 제공하지 않긴해서 이대로 둬도 괜찮을것같긴합니다만, 추가할 수 있게 주석으로 코드 추가해놨습니다!

Moderator11 commented on src/pages/login/LoginPage.tsx:
```diff
@@ -1,5 +1,121 @@
+  cursor: pointer;
+  text-decoration: underline;
+  font-weight: 600;
+  font-size: inherit;
```
와우, 이런 속성이 있는지 처음 알았네요 :0 잘 배워갑니당

Moderator11 commented on src/components/InputField.tsx:
개인적으로 InputField는 자주 겹칠 수 있는 평범한 이름이라고 생각해요, 입력 기능에 더해 유효성 검사, 에러 메세지 출력등의 부가 기능들이 합쳐진 컴포넌트이니 ValidationInputField, AdvancedInputField, FormInputField와 같이 이 컴포넌트의 속성을 잘 설명할 수 있는 접두사를 붙이는건 어떨까요?

seongjin0320 replied:
폼에 사용되는 인풋필드이므로 FormInputField로 이름 변경했습니다~

Moderator11 reviewed (approved):
@seongjin0320 안녕하세요! 리뷰어 박수민입니다 :)

디자인 깔끔한거 너무 좋아요 :0
파일 분리랑 공통 컴포넌트 뽑아낸 것도 너무 잘 하신 것 같아요! 유효성 검증할 수 있는 InputField가 개인 스페이스 페이지에도 필요했습니다ㅎㅎ

리뷰와 함께 몇가지 질문들 남겼는데 답해주시면 감사하겠습니다.
크게 수정 필요한 부분은 없는 것 같아 Approve 해놓겠습니다.

로그인/회원가입 페이지 구현 너무 수고하셨습니다!

Moderator11 commented on src/pages/register/RegisterPage.tsx:
```diff
@@ -1,5 +1,149 @@
+          placeholder="예: INFP"
+          register={register}
+          error={errors.mbti}
+        />
```
<img width="395" height="92" alt="image" src="https://github.com/user-attachments/assets/c865ce5c-4f6f-4b72-8083-186f260d5013" />

MBTI 부분을 텍스트 입력 형식으로 만들고 유효성 검사로 로직을 작성하셨네요!
이부분은 드롭다운 선택 필드로 바꾸면 UX 경험이 더 좋지 않을까요?

seongjin0320 replied:
드롭다운 선택 필드로 변경했습니다!

whyyhyh reviewed (approved):
안녕하세요!! 남규리입니다.!!
로그인, 회원가입 창의 디자인적 요소도 통일성 있고 전체 디자인과 어우러지게 잘 만드신 것 같아요! 필요한 요소들도 잘 들어있구요!
너무 수고하셨습니당👍👍
수민님께서 이미 리뷰를 너무 잘 해주셨고 제가 아직 코드 보는 눈이 좋지 못해서 코멘트를 별로 남기지 못했습니다 ㅎㅎ ㅜ
첫 주차 너무 수고하셨습니당~

seongjin0320 committed:
feat: 로그인, 회원가입 유효성검사 강화 로직 주석 추

- 필요할 때 강화로직을 사용할 수 있습니다.

seongjin0320 committed:
refactor: 피드백 반영

- 공통컴포넌트 InputField -> FormInputField로 이름 변경
- 공통컴포넌트 FormSelectField 추가
- 회원가입시 MBTI 입력란 드롭다운 선택 필드 방식으로 변경

Moderator11 commented:
수정 사항 대부분을 반영해주셨네요! 너무 수고하셨습니다 :)
> 의도한것은 아니긴했는데, 생각해보니 실제 다른 서비스들에서도 로그인시에 비밀번호의 자세한 유효성 피드백을 제공하지 않긴해서 이대로 둬도 괜찮을것같긴합니다만, 추가할 수 있게 주석으로 코드 추가해놨습니다!

오 저도 생각해보니 다른 웹사이트에서 로그인시에는 복잡한 유효성 규칙에 대한 오류 메세지가 없었던 것 같아욥 👍
